expression/expression.go

```diff
@@ -427,12 +427,9 @@ func toBool(sc *stmtctx.StatementContext, eType types.EvalType, buf *chunk.Colum
 			if buf.IsNull(i) {
 				isZero[i] = -1
 			} else {
-				iVal, err := types.StrToInt(sc, buf.GetString(i))
+				iVal, err := types.StrToFloat(sc, buf.GetString(i))
 				if err != nil {
-					iVal, err = HandleOverflowOnSelection(sc, iVal, err)
-					if err != nil {
-						return err
-					}
```

Comment:
Function `types.StrToFloat` does not need to handle `overflow`. So the test `TestVecToBool` is unused.